draft-irtf-cfrg-vdaf.md

```diff
@@ -1190,6 +1194,190 @@ how the nonces are chosen, but security requires that the nonces be unique. See
 execution of a VDAF requires the application to instantiate secure channels
 between each of the protocol participants.
 
+## Communication Patterns for Preparation {#vdaf-prep-comm}
+
+In each round of preparation, each Aggregator writes a prep share to the
+channel, which is then processed into the prep message using the public
+`prep_shares_to_prep` algorithm and broadcast to the Aggregators to start the
+next round. In this section we describe some approaches to realizing this
+braodcast channel functionality in protocols that use VDAFs.
+
+The state machine of each Aggregator for VDAF preparation is shown in
+{{vdaf-prep-state-machine}}.
+
+~~~ state
+ START --> CONTINUED(round, host_prep_state)
+  |                |                    |
+  |                |                    v
+  +--> REJECTED <--+     FINISHED(out_share)
+~~~
+{: #vdaf-prep-state-machine title="State machine for VDAF preparation."}
+
+State transitions are made when the state is acted upon by the host's local
+inputs and/or messages sent by the peers. The initial state is `START`. The
+terminal states are `REJECTED`, indicating that the report was rejected and
+cannot be processed any further; and `FINISHED(out_share)` where `out_share` is
+the Aggreagtor's recovered output share.
+
+The first state transition is induced by the following transition rule:
+
+~~~ transition
+State START
+
+def prep_start(Vdaf,
+               vdaf_verify_key: bytes[Vdaf.VERIFY_KEY_SIZE],
+               is_leader: bool,
+               agg_param: Vdaf.AggParam,
+               nonce: bytes[Vdaf.NONCE_SIZE],
+               public_share: bytes,
+               host_input_share: bytes):
+    try:
+        host_prep_state = Vdaf.prep_init(
+            vdaf_verify_key,
+            0 if is_leader else 1,
+            agg_param,
+            nonce,
+            public_share,
+            host_input_share,
+        )
+        state CONTINUED(0, host_prep_state)
```

Comment:
Reviewer note: This obviously is not grammatically correct Python. Our goal thus far was to have all pseudocode in the draft be runnable, so we might consider how to do so neatly. However in the interest of moving DAP-05 forward, I would prefer to not be too nit-picky, at least for the moment.